code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+
+		if(length(candidates) && !host.mind) //check if anyone wanted to play as the dead person and check if no one's in control of the body one last time.
+			var/mob/dead/observer/ghost = pick(candidates)
+			host.key = ghost.key
+			host.suiciding = 0 //turns off the suicide var just in case
+			host.revive(full_heal = TRUE, admin_revive = TRUE) //might as well go all the way
+			to_chat(host, "<span class='warning'>You may wear the skin of someone else, but you know who and what you are. Pretend to be the original owner of this body as best as you can.</span>")
+			to_chat(user, "<span class='notice'>[host.name] is slowly getting back up with an empty look in [host.p_their()] eyes. It...worked?</span>")
+			playsound(host, "sound/magic/castsummon.ogg", 50, TRUE)
+			return
+
+		else
+			to_chat(user, "<span class='warning'>The rune failed! Maybe you should try again later.</span>")
+
+
+/* black space rune : will swap out the species of the two next person walking on the rune  */
+
+
+/obj/item/slimecross/warping/black
+	colour = "black"
+	runepath = /obj/effect/warped_rune/blackspace
+	effect_desc = "Will swap the species of the first two humanoids that walk on the rune. Also works on corpses."
+	drawing_time = 100
+
+
+/obj/effect/warped_rune/blackspace
+	icon_state = "cursed_rune"
+	desc = "Your body is the problem, limited, so very very limited."
+	///first person to step on the rune
+	var/mob/living/carbon/human/first_person
+	///second person to step on the rune
+	var/mob/living/carbon/human/second_person
+	///here to check if someone already stepped on the rune
+	var/stepped_on = FALSE
+
+
+/obj/effect/warped_rune/blackspace/Initialize()
+	. = ..()
+	cooldown = 0 //doesn't start on cooldown like most runes
+
+
+///will swap the species of the first two human or human subset that walk on the rune
+/obj/effect/warped_rune/blackspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(cooldown > world.time) //here to avoid spam/lag
+		to_chat(crossing, "<span class='warning'>The rune needs a little more time before processing your DNA!</span>")
+		return
+
+	if(!istype(crossing,/mob/living/carbon/human))
+		return
+
+	var/mob/living/carbon/human/crosser = crossing
+	if(!stepped_on)
+		first_person = crosser
+		stepped_on = TRUE
+		return
+
+	if(crosser == first_person || crosser.stat == DEAD)
+		return
+
+	second_person = crosser
+	var/first_dna = first_person.dna.species
+	var/second_dna = second_person.dna.species
+	second_person.set_species(first_dna)  //swap the species
+	first_person.set_species(second_dna)
+	stepped_on = FALSE
+	cooldown = max_cooldown + world.time //the default max cooldown is of 10 seconds
+
+/obj/effect/warped_rune/blackspace/Destroy()
+	first_person = null
+	second_person = null
+	return ..()
+
+
+/*Acts as a bunker making anything into the rune immune to outside explosions and gas leaks. doesn't work if the epicenter of the explosion is IN the rune */
+
+
+/obj/item/slimecross/warping/oil
```

Comment:
could you maybe make this protect a wider area, so you can use this to "ward" departments against explosions/plasma fires?

Reply:
cool rune, though